### DISCUSSION TITLE
impossible to build with vitejs

### DISCUSSION THREAD

**Discussion by wadouk** [vfile/vfile#65] [Q&A] [answered]
True to build unified with vite and it fail at runtime because of `process.cwd()` used without protection.
It should be great to have a `pre-built` version for browser because it's unusable without a bundler that override `process`.

**Comment by wadouk** [comment]
had to patch at beginning of a file 

```js
window.process = {
    cwd: () => '/'
}
```

use version 4.2

will try with an update

**Comment by ChristianMurphy** [comment, marked as answer]
@wadouk what version of `vfile` are you using?
In version 5 and higher, `vfile` provides https://github.com/vfile/vfile/blob/b4c9fb85e975be2b32e03e47c4bc8a1501155493/package.json#L38-L42 which removes the dependency on `process`, `path`, and some node specific `url` features for browser builds.
If you are using version 5 and `vite` is ignoring the `browser` field, it may be an issue to raise with `vite`, it appears that `vite` should support it (https://github.com/vitejs/vite/blob/fa6475fd4230fb2ab200485bc35b136b2474abfd/packages/vite/src/node/plugins/resolve.ts#L635-L640)

In the meantime you could try https://github.com/snowpackjs/rollup-plugin-polyfill-node

**Reply by wadouk** [reply]
Thanks, seen that. 
Try to upgrade `unified` to have a more up to date version of vfile but some mandatory used plugin are not upgraded.
[Partial upgrade with yarn](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/) to `vfile@5` doesn't work because `unified@9.1.0` specify `vfile@4`

Think will go back to my workaround and handle those later.

**Reply by ChristianMurphy** [reply]
right, `vfile` 5 should be used along side `unified` 10, and `vfile` 4 with `unified` 9.
